django/db/models/aggregates.py

```diff
@@ -36,6 +37,57 @@ def default_alias(self):
     def get_group_by_cols(self):
         return []
 
+    def filter(self, **kwargs):
+        filter_q = Q(**kwargs)
+        return AggregateFilter(self.copy(), filter_q)
+
+    def exclude(self, **kwargs):
+        filter_q = ~Q(**kwargs)
+        return AggregateFilter(self.copy(), filter_q)
+
+
+class AggregateFilter(Expression):
+    def __init__(self, value, condition, output_field=None):
+        super().__init__(output_field=output_field)
+        self.value = value
+        self.condition = condition
+
+    def get_source_expressions(self):
+        return [self.value]
+
+    def set_source_expressions(self, exprs):
+        self.value = exprs[0]
+
+    def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
+        c = self.copy()
+        c.value = c.value.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        c.condition = c.condition.resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        c.old_condition = self.condition
+        return c
+
+    def as_sql(self, compiler, connection):
+        field = self.value.get_source_expressions()[0]
+
+        x = Case(
+            When(
+                self.condition, then=field
+            )
+        )
+
+        self.value.set_source_expressions([x])
+        return compiler.compile(self.value)
+
+    def as_postgresql(self, compiler, connection):
+        from django.contrib.postgres.aggregates import FilterWhere
+
+        return compiler.compile(FilterWhere(self.value, self.condition, output_field=self.output_field))
+
+    def filter(self, **kwargs):
+        return AggregateFilter(value=self.value, condition=self.condition | Q(**kwargs))
```

Comment:
This will also have to take into account `filter(m2m__field=val1, criteria=m2m__otherfield=val2)` != `filter(m2m__field=val1).filter(m2m__field=val2)` as explained in [spanning multi-valued relationships](https://docs.djangoproject.com/en/1.11/topics/db/queries/#spanning-multi-valued-relationships)